src/utils/firebase/auth.ts

```diff
@@ -13,6 +15,23 @@ import { firebaseEmulatorConfig } from "../../constants/firebase";
 
 const firebaseAuth = getAuth(firebaseApp);
 
+let currentUser: User | null | undefined = undefined;
+
+onAuthStateChanged(firebaseAuth, (user) => (currentUser = user));
+
+export const loadCurrentUser = (): Promise<User | null> =>
+  new Promise((resolve) => {
+    const checkVariable = () => {
+      if (currentUser !== undefined) {
+        resolve(currentUser);
+      } else {
+        setTimeout(checkVariable, 100);
+      }
+    };
+
+    checkVariable();
+  });
+
```

Comment:
```suggestion
export const loadCurrentUser = (): Promise<User | null> =>
  new Promise((resolve) => {
    const unsubscribe = onAuthStateChanged(firebaseAuth, (user) => {
      resolve(user);
      unsubscribe();
    });
  });
```

Will this be better? It seems like your origin method will keep listening every 100ms. Which might let efficacy go down.

Reply:
Updated. Thanks